src/meta_schedule/utils.h

```diff
@@ -50,6 +51,201 @@
+                      << loop->extent->GetTypeKey();
+      result *= int_imm->value;
+      return result;
+    }
+
+#define TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(Node) \
+  TResult VisitExpr_(const Node* op) final {        \
+    TResult result(op->dtype);                      \
+    result += VisitExpr(op->a);                     \
+    result += VisitExpr(op->b);                     \
+    return result;                                  \
+  }
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(AddNode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(SubNode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(MulNode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(DivNode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(ModNode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(FloorDivNode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(FloorModNode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(MinNode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(MaxNode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(EQNode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(NENode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(LTNode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(LENode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(GTNode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(GENode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(AndNode);
+    TVM_META_SCHEDULE_FLOP_COUNTER_BINARY(OrNode);
+#undef TVM_META_SCHEDULE_FLOP_COUNTER_BINARY
+    TResult VisitExpr_(const CastNode* op) override { return VisitExpr(op->value); }
+    TResult VisitExpr_(const VarNode* op) override { return TResult(); }
+    TResult VisitExpr_(const SizeVarNode* op) override { return TResult(); }
+    TResult VisitExpr_(const BufferLoadNode* op) override { return TResult(); }
+    TResult VisitExpr_(const IntImmNode* op) override { return TResult(); }
+    TResult VisitExpr_(const FloatImmNode* op) override { return TResult(); }
+    TResult VisitExpr_(const NotNode* op) override {
+      TResult result(op->dtype);
+      result += VisitExpr(op->a);
+      return result;
+    }
+    TResult VisitExpr_(const SelectNode* op) override {
+      TResult cond = VisitExpr(op->condition);
+      cond += VisitExpr(op->true_value).MaxWith(VisitExpr(op->false_value));
+      return cond;
+    }
+    TResult VisitExpr_(const CallNode* op) override {
+      TResult ret;
+      for (const auto& x : op->args) {
+        ret += VisitExpr(x);
+      }
+      return ret;
+    }
+  };
+  FlopCounter counter;
+  TResult result;
+  for (const auto& kv : mod->functions) {
+    const BaseFunc& base_func = kv.second;
+    if (const auto* prim_func = base_func.as<PrimFuncNode>()) {
+      result += counter.VisitStmt(prim_func->body);
+    }
+  }
+  double cnt = 0.0;
+  int i32 = TResult::DataType2Int(tvm::DataType::Int(32));
+  int i64 = TResult::DataType2Int(tvm::DataType::Int(64));
+  int u1 = TResult::DataType2Int(tvm::DataType::UInt(1));
+  for (const auto& kv : result.data_) {
+    if (kv.first != i32 && kv.first != i64 && kv.first != u1) {
```

Comment:
Can we remove `kv.first != i32`? Right now when I tune int8 workloads, whose output is always int32, I get 0 GFLOPS like

```
Trial #0: GFLOPs: 0.0000. Time: 0.0576 ms. Best GFLOPs: 0.0000
```

Reply:
CC @junrushao1994

Reply:
We will lift the method as a standalone analysis function and expose it for more general usage. Will work on that after @zxybazh’s vacation